ReactNativeClient/root.js

```diff
@@ -660,6 +624,54 @@ class AppComponent extends React.Component {
 		});
 
 		AppState.addEventListener('change', this.onAppStateChange_);
+
+		const sharedData = Platform.OS === 'android' ? await ShareExtension.data() : null;
+		if (sharedData) {
+			this.handleShared(sharedData);
+		}
+	}
+
+	async handleShared(sharedData) {
+		reg.logger().info(`Received shared data: ${JSON.stringify(sharedData)}`);
+
+		if (!this.props.selectedFolderId) return;
+
+		if (!!sharedData.resources && sharedData.resources.length > 0) {
+			let granted = await check(PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE);
+			if (granted !== RESULTS.GRANTED) {
+				granted = await request(PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE);
+				if (granted !== RESULTS.GRANTED) {
+					ToastAndroid.show('Cannot receive shared data - permission denied', ToastAndroid.SHORT);
+					return;
+				}
+			}
+		}
+
+		// This is a bit hacky, but the surest way to go to
+		// the needed note. We go back one screen in case there's
+		// already a note open - if we don't do this, the dispatch
+		// below will do nothing (because routeName wouldn't change)
+		// Then we wait a bit for the state to be set correctly, and
+		// finally we go to the new note.
+		await this.props.dispatch({ type: 'NAV_BACK' });
+
+		await this.props.dispatch({ type: 'SIDE_MENU_CLOSE' });
+
+		const newNote = await Note.save({
+			parent_id: this.props.selectedFolderId,
+		}, { provisional: true });
+
+		setTimeout(() => {
+			this.props.dispatch({
+				type: 'NAV_GO',
+				routeName: 'Note',
+				noteId: newNote.id,
+				sharedData: {
```

Comment:
Why isn't this just `sharedData: sharedData`? Is there some restriction on `ShareExtension.data()`?

Reply:
Thanks, good catch. This piece went though quite a few changes and I lost track.

Reply:
Fxied